File: src/main/java/com/eceplatform/QAForum/Actors/MailActor.java
package com.eceplatform.QAForum.Actors;

import akka.actor.AbstractLoggingActor;
import org.springframework.mail.SimpleMailMessage;
import org.springframework.mail.javamail.JavaMailSender;

public class MailActor extends AbstractLoggingActor {

    private JavaMailSender javaMailSender;

    public MailActor(JavaMailSender javaMailSender) {
        this.javaMailSender = javaMailSender;
    }

    @Override
    public Receive createReceive() {
        return receiveBuilder()
                .match(
                        SimpleMailMessage.class,
                        smm -> {
                            log().info("Mail message to be sent: {}", smm);
                            try {
                                javaMailSender.send(smm);
                                log().info("Mail sent successfully: {}", smm);
                            } catch (Exception e) {
                                log().error("Error in sending mail: {} for {}", e, smm);
                            }
                        })
                .matchAny(o -> log().error("received unknown message"))
                .build();
    }
}

File: src/main/java/com/eceplatform/QAForum/actors/MailActor.scala
package com.eceplatform.QAForum.actors

import akka.actor.{Actor, ActorLogging}
import org.springframework.mail.SimpleMailMessage
import org.springframework.mail.javamail.JavaMailSender

case class MailActor(var javaMailSender: JavaMailSender) extends Actor with ActorLogging {
  override def receive: Receive = {
    case smm: SimpleMailMessage => {
      log.info("Mail message to be sent: {}", smm)
      try {
        javaMailSender.send(smm)
        log.info("Mail sent successfully: {}", smm)
      } catch {
        case e: Exception =>
          log.error("Error in sending mail: {} for {}", e, smm)
      }
    }
    case _ => log.error("received unknown message");
  }
}